test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
+    execute('autocmd TextYankPost * let g:event = copy(v:event)')
```

Comment:
Current implementation *replaces dictionaries*. You suggest to *clear keys*. I am not saying that one variant should be preferred over another, but currently one may save `v:event` somewhere just by using `let foo = v:event` and I suggest to test this, and adjust tests to protect against changing this behaviour, thus making such change intentional.

Clearing keys should reduce number of memory allocations and frees. Replacing dictionaries like now is less code to write (by one function: `dict_clear` which is absent).